src/Sets/Interval.jl

```diff
@@ -514,3 +514,21 @@ We consider the interval as a line segment with y coordinate equal to zero.
 function plot_recipe(I::Interval{N}, ε::N=zero(N)) where {N<:Real}
     return [min(I), max(I)], zeros(N, 2)
 end
+
+"""
+    scale(α::Real, x::Interval)
+
+Concrete scaling of an interval.
+
+### Input
+
+- `α` -- scalar
+- `x` -- interval
+
+### Output
+
+The interval obtained by applying the numerical scale to the given interval.
+"""
+function scale(α::Real, x::Interval)
```

Comment:
True, the library in most (all?) cases restricts the same numeric types for binary operations like these, let's do that here too. 

On the other hand, i don't like that restriction; it seems unnatural for new users because Julia handles the type promotion automatically (try `1` times `[1.]`). Eventually the library should do the same that too in my opinion, iI would like to lift that restriction and there are open issues related to that (https://github.com/JuliaReach/LazySets.jl/issues/1311, https://github.com/JuliaReach/LazySets.jl/issues/1249, https://github.com/JuliaReach/LazySets.jl/issues/1181, https://github.com/JuliaReach/LazySets.jl/issues/733).

Reply:
I made the change, so i think that this conversation is resolved. Note that before the change, the automatic promotion happens in `α * x.dat`.